components/create_comment/index.js

```diff
@@ -76,6 +74,7 @@ function makeMapStateToProps() {
             permission: Permissions.USE_GROUP_MENTIONS,
         });
         const channelMemberCountsByGroup = selectChannelMemberCountsByGroup(state, ownProps.channelId);
+        const groupsWithAllowReference = useGroupMentions ? getAssociatedGroupsForReferenceByMention(state, channel.team_id, channel.id) : new Map();
```

Comment:
We can't pass in a new object when we're not using group mentions here or else CreateComment will constantly re-render unnecessarily. It did that before with the previous code as well, but we should fix it while we're in here. Instead, we should pass in `null` and handle it in the component.

Reply:
Updated. Thanks.